lib/rules/no-extra-boolean-cast.js

```diff
@@ -110,6 +158,10 @@ module.exports = {
                                 prefix = " ";
                             }
 
+                            if (astUtils.getPrecedence(node.argument) < astUtils.getPrecedence(parent)) {
```

Comment:
This should be `< astUtils.getPrecedence(parent.parent)` because we're replacing parent here (which is always a `!` node).

So this might leave some unnecessary parens when the precedence is somewhere between the precedence of `!` and the precedence of the `!!`s parent, e.g.:

```js
if (!!(a + b) && c) {
    // ...
}
```

Though there are already some cases with unnecessary parens regardless of this, but that should be fixed in a separate PR.